src/main/java/com/minecolonies/apiimp/initializer/InteractionValidatorInitializer.java

```diff
@@ -196,10 +201,22 @@ public static void init()
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(FURNACE_USER_NO_FUEL),
           citizen -> citizen.getWorkBuilding() instanceof AbstractBuildingFurnaceUser && ((AbstractBuildingFurnaceUser) citizen.getWorkBuilding()).getAllowedFuel().isEmpty());
         InteractionValidatorRegistry.registerStandardPredicate(new TranslationTextComponent(FURNACE_USER_NO_FOOD),
-          citizen -> citizen.getWorkBuilding() instanceof BuildingCook && (IColonyManager.getInstance().getCompatibilityManager().getFood().stream()
-                  .filter(storage -> !((AbstractFilterableListBuilding) citizen.getWorkBuilding()).getCopyOfAllowedItems().get("food").contains(storage))
-                  .map(ItemStorage::getItemStack)
-                  .collect(Collectors.toList())).isEmpty());
+          citizen -> {
+            if (!(citizen.getWorkBuilding() instanceof BuildingCook))
+            {
+                return false;
+            }
+
+            for (final ItemStorage storage : IColonyManager.getInstance().getCompatibilityManager().getEdibles())
+            {
+                if (!citizen.getWorkBuilding().getModule(GroupedItemListModule.class).map(m -> m.getList(FOOD_EXCLUSION_LIST).contains(storage)).orElse(false))
+                {
+                    return false;
+                }
+            }
```

Comment:
Flip this around.
Get the module first and then in the map of the optional do the loop.
Much more efficient.